docs/user_guide/modeling_extensions/uncertainty_propagation/index.rst

```diff
@@ -0,0 +1,132 @@
+
+All gradients are calculated with `k_aug <https://github.com/dthierry/k_aug>`_  [1]. More specifically, :math:`\frac{\partial f}{\partial p}, \frac{\partial f}{\partial x}, \frac{\partial c_1}{\partial p}, \frac{\partial c_1}{\partial x}, \ldots, \frac{\partial c_k}{\partial p}, \frac{\partial c_k}{\partial x}`, are computed via automatic differentiation whereas :math:`\frac{\partial x}{\partial p}` are computed via nonlinear programming sensitivity theory.
+
+The covariance matrix :math:`\Sigma_p` is either user supplied or obtained via regression (with ParmEst). 
+
+**Dependencies**
+
+`k_aug <https://github.com/dthierry/k_aug>`_ [1] is required to use uncertainty_propagation module. If you have the IDAES framework installed, you can get k_aug with the IDAES extensions by running the following command:
+
+.. code-block:: python
+
+   idaes get-extensions
+
+Basic Usage
+------------
+The ``uncertainty_propagation``'s main function is **quantify_propagate_uncertainty**. The quantify_propagate_uncertainty python function can have eight arguments and returns a namedtuple output. 
+
+The following example shows a usage of **quantify_propagate_uncertainty** with a Rooney and Biegler model [2].
+
+.. code:: python
+
+   # Required imports
+   >>> from idaes.apps.uncertainty_propagation.uncertainties import quantify_propagate_uncertainty
+   >>> import pyomo.environ as *
+   >>> import pandas as pd
+
+   # Define rooney_biegler_model
+   >>> def rooney_biegler_model(data):
+           model = ConcreteModel()
+           model.asymptote = Var(initialize = 15)
+           model.rate_constant = Var(initialize = 0.5)
+           
+           def response_rule(m, h):
+               expr = m.asymptote * (1 - exp(-m.rate_constant * h))
+               return expr
+           model.response_function = Expression(data.hour, rule = response_rule)
+    
+           def SSE_rule(m):
+               return sum((data.y[i] - m.response_function[data.hour[i]])**2 for i in data.index)
+           model.SSE = Objective(rule = SSE_rule, sense=minimize)
+           return model
+
+   # Define rooney_biegler_model_opt
+   >>> def rooney_biegler_model_opt(data):
+           model = ConcreteModel()
+           model.asymptote = Var(initialize = 15)
+           model.rate_constant = Var(initialize = 0.5)
+           model.obj = Objective(expr=model.asymptote*(1-exp(-model.rate_constant*10))
+                                 , sense=minimize)
+           return model
+   
+   # Define data
+   >>> data = pd.DataFrame(data=[[1,8.3],[2,10.3],[3,19.0],[4,16.0],[5,15.6],[7,19.8]],
+                           columns=['hour', 'y'])
+
+   # Define variable_name
+   >>> variable_name = ['asymptote', 'rate_constant']
+
+   # Define obj_function
+   >>> def obj_function(model, data):
```

Comment:
Rename this function `SSE_obj_function`. That will be more clear. We have two optimization problems, so we want to be more explicit about the name.

Reply:
Changed them.